### ISSUE TITLE
Firebase-Amplify dependency conflicts

### ISSUE THREAD

**Issue by exampaqs** [aws-amplify/amplify-flutter#434]
And because exampaqs depends on amplify_storage_s3 ^0.1.0 which depends on plugin_platform_interface ^1.0.1, firebase_auth
  ^1.0.0-1.0.nullsafety.0 requires firebase_core >=0.8.0-1.0.nullsafety.2 <0.8.0-2.0.nullsafety.0.
So, because exampaqs depends on both firebase_auth ^1.0.0-1.0.nullsafety.0 and firebase_core ^1.0.0-1.0.nullsafety.0, version solving
  failed.

because i upgraded firebase version and other dependencies in my pubspec.yaml file, i get the error above in my console because my project also s3 bucket and cognito-auth heavily. And when i comment out all amplify packages, the code works, so I feel there is a need for amplify upgrade asap.

Firebase depends on firebase_core_platform_interface 4.0.0 while amplify depends on  plugin_platform_interface ^1.0.1

**Comment by fjnoyp** [comment]
Hi @exampaqs thanks for sharing your issue here. 

Could you provide your full pubspec.yaml so we can investigate all your dependencies?

**Comment by exampaqs** [comment]
> Hi @exampaqs thanks for sharing your issue here.
> 
> Could you provide your full pubspec.yaml so we can investigate all your dependencies?


---------------------------------------.yaml file---------------------------------------------------
version: 1.0.0+1

environment:
  sdk: ">=2.10.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  # amplify_analytics_pinpoint: ^0.1.0
  # amplify_auth_cognito: ^0.1.0
  # amplify_flutter: ^0.1.0
  amplify_storage_s3: ^0.1.0
  # cached_network_image: ^2.5.0
  cloud_firestore: ^1.0.0-1.0.nullsafety.0
  connectivity: ^3.0.2
  countdown: ^0.1.0
  expandable: ^4.1.4
  firebase_auth: ^1.0.0-1.0.nullsafety.0
  firebase_core: ^1.0.0-1.0.nullsafety.0
  flutter_bloc: ^7.0.0-nullsafety.0
  flutter_countdown_timer: ^4.0.0
  flutter_html: ^2.0.0-nullsafety.0
  flutter_launcher_icons: ^0.9.0
  flutter_svg: ^0.21.0-nullsafety.0
  hexcolor: ^2.0.0
  http: ^0.13.0-nullsafety.0
  pdf: ^3.0.0-nullsafety.1
  provider: ^5.0.0-nullsafety.5 
  responsive_grid: ^1.2.1+1
  shared_preferences: ^2.0.3

dev_dependencies:
  flutter_test:
    sdk: flutter

**Comment by hysuki** [comment]
@fjnoyp Similar dependency conflicts issue when trying to use with Hive 2.0.0

```bash
Because no versions of amplify_api match >0.1.0 <0.2.0 and amplify_api 0.1.0 depends on amplify_api_plugin_interface 0.1.0, amplify_api ^0.1.0 requires amplify_api_plugin_interface 0.1.0.
And because amplify_api_plugin_interface 0.1.0 depends on uuid ^2.2.2, amplify_api ^0.1.0 requires uuid ^2.2.2.
Because uuid 2.2.2 depends on crypto ^2.0.0 and no versions of uuid match >2.2.2 <3.0.0, uuid ^2.2.2 requires crypto ^2.0.0.
Thus, amplify_api ^0.1.0 requires crypto ^2.0.0.
And because hive >=2.0.0 depends on crypto ^3.0.0, amplify_api ^0.1.0 is incompatible with hive >=2.0.0.
So, because airwith depends on both hive ^2.0.0 and amplify_api ^0.1.0, version solving failed.
pub get failed (1; So, because airwith depends on both hive ^2.0.0 and amplify_api ^0.1.0, version solving failed.)
```

```yaml
dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^1.0.2
  get: ^3.26.0
  logger: ^0.9.4
  hive: ^2.0.0
  hive_flutter: ^1.0.0

  # amplify
  amplify_flutter: ^0.1.0
  amplify_auth_cognito: ^0.1.0
  amplify_analytics_pinpoint: ^0.1.0
  amplify_storage_s3: ^0.1.0
  amplify_api: ^0.1.0

dev_dependencies:
  flutter_test:
    sdk: flutter
  hive_generator: ^1.0.0
  build_runner: ^1.11.5
```

**Comment by kay-si** [comment]
Similar one

```
% flutter pub get
Because firebase_auth >=1.0.0 depends on firebase_auth_platform_interface ^4.0.0 which depends on plugin_platform_interface ^2.0.0, firebase_auth >=1.0.0 requires plugin_platform_interface ^2.0.0.
And because every version of amplify_flutter depends on plugin_platform_interface ^1.0.1, firebase_auth >=1.0.0 is incompatible with amplify_flutter.
So, because kanri_online_app depends on both firebase_auth ^1.0.0 and amplify_flutter <1.0.0, version solving failed.
```

```
dependencies:
  flutter:
    sdk: flutter

  cupertino_icons: ^1.0.0
  webview_flutter: ^1.0.7
  image_picker: ^0.7.2
  url_launcher: ^6.0.2
  logging: ^0.11.4
  flutter_user_agent: ^1.2.2
  google_sign_in: ^4.5.9
  connectivity: ^3.0.2
  webview_cookie_manager: ^1.0.8
  firebase_core: ^1.0.0
  firebase_auth: ^1.0.0
  firebase_messaging: ^9.0.0
  firebase_crashlytics: ^1.0.0
  amplify_flutter: '<1.0.0'
  amplify_auth_cognito: '<1.0.0'
  amplify_storage_s3: '<1.0.0'
  crypto: ^2.1.5
  image_downloader: ^0.20.1
  uni_links: ^0.4.0

dev_dependencies:
  flutter_test:
    sdk: flutter
  flutter_launcher_icons: "^0.7.0"
```

**Comment by PaoloP98** [comment]
Same

**Comment by FranWeidt** [comment]
Similar issue when trying to use with `localizely_sdk: ^2.0.0`. Localizely depends on `uuid ^3.0.0` and amplify uses an older version `uuid^2.2.2`.

```
Because localizely_sdk >=2.0.0 depends on uuid ^3.0.0 and amplify_api_plugin_interface 0.1.0 depends on uuid ^2.2.2, localizely_sdk >=2.0.0 is incompatible with amplify_api_plugin_interface 0.1.0.

And because amplify_flutter 0.1.0 depends on amplify_api_plugin_interface 0.1.0 and no versions of amplify_flutter match >0.1.0 <0.2.0, localizely_sdk >=2.0.0 is incompatible with amplify_flutter ^0.1.0.

```

**Comment by fjnoyp** [comment]
Hi everyone, thank you for posting your issues here.  We are actively investigating a fix for this.

**Comment by fjnoyp** [comment]
Hi everyone, we have a temporary fix that should get you unblocked for now as we work on a longer term solution.  


@exampaqs @hysuki @kay-si @PaoloP98:
You are having an issue with using Firebase because we use different versions of plugin_platform_interface.  Adding the following to your pubspec.yaml should resolve your issue: 
`
dependency_overrides:
  plugin_platform_interface: ^2.0.0
`


@FranWeidt:
Adding the following to your pubspec.yaml should resolve your issue: 
`
dependency_overrides:
  uuid: '3.0.1'
`

**Comment by krunduev** [comment]
Because every version of amplify_flutter depends on plugin_platform_interface ^1.0.1 and flutter_local_notifications_platform_interface >=3.0.0-nullsafety.3 depends on plugin_platform_interface ^2.0.0, amplify_flutter is incompatible with flutter_local_notifications_platform_interface >=3.0.0-nullsafety.3.
And because flutter_local_notifications >=5.0.0 depends on flutter_local_notifications_platform_interface ^3.0.0, amplify_flutter is incompatible with flutter_local_notifications >=5.0.0.

So, because xxxxxx depends on both **flutter_local_notifications ^5.0.0** and **amplify_flutter ^0.1.0**, version solving failed.

**Comment by krunduev** [comment]
Also I cant use dependency_overrides because so many libraries already migrate to Flutter 2.0 
ANd I get conflict not only with **flutter_local_notifications**

**Comment by LUCIFER-dev-king** [comment]
@krunduev Having the same issue.

**Comment by SalahAdDin** [comment]
It happens with `uuid` too:
```shell
[thesis_cancer] flutter pub get
Running "flutter pub get" in thesis_cancer...                   
Because freezed >=0.14.1+1 depends on build ^2.0.0 which depends on convert ^3.0.0, freezed >=0.14.1+1 requires convert ^3.0.0.

Because uuid 2.2.2 depends on convert ^2.0.0 and no versions of uuid match >2.2.2 <3.0.0, uuid ^2.2.2 requires convert ^2.0.0.

Thus, freezed >=0.14.1+1 is incompatible with uuid ^2.2.2.

And because amplify_api_plugin_interface 0.1.0 depends on uuid ^2.2.2, freezed >=0.14.1+1 is incompatible with amplify_api_plugin_interface 0.1.0.

And because amplify_flutter 0.1.0 depends on amplify_api_plugin_interface 0.1.0 and no versions of amplify_flutter match >0.1.0 <0.2.0, freezed >=0.14.1+1 is incompatible with amplify_flutter ^0.1.0.

So, because thesis_cancer depends on both amplify_flutter ^0.1.0 and freezed ^0.14.1+1, version solving failed.
pub get failed (1; So, because thesis_cancer depends on both amplify_flutter ^0.1.0 and freezed ^0.14.1+1, version solving failed.)
exit code 1
```

**Comment by LUCIFER-dev-king** [comment]
@krunduev You can use ``` flutter_local_notification: 4.0.0 ``` This version is compatible with amplify.

**Comment by haverchuck** [comment]
@exampaqs This should be resolved by the 0.1.1 release.
